packages/clay-core/src/picker/Picker.tsx

```diff
@@ -126,6 +126,11 @@ export type Props<T> = {
 	 */
 	selectedKey?: React.Key;
 
+	/**
+	 * Flag to define the panel width behavior.
+	 */
+	width?: 'flexible' | 'fixed';
```

Comment:
@pat270 according to the documentation it says that the minimum should still be 160px even with flexible width. Now it seems that there is an exception that is for the case of year selection from what I understand, in this case I am thinking about allowing you to define the width number in the API or just remove the limitation when it is flexible so it can be used in common sense. What do you think?

Reply:
I don't think a fixed width is that useful these days, but we can go ahead and see what happens? I think being able to set a min and max width is more future proof.

Reply:
The big problem with going with max and min width is that this adds a lot of flexibility that will allow things that Lexicon doesn't foresee, for example `flexible` always has to respect the width size of the trigger if we add max width that would be a problem. This would also be a problem for responsive because we would have to create a mechanism to identify this and it would make things more complex.

The minimum needs to be respected at least for fixed cases and this is in CSS so it seems ok, when it is `flexible` the minimum also needs to be 160px but as I said there is an exception so I'm not sure if I should set this limit or just leave it free following the width of the trigger and let them use common sense as described in the documentation, which in flexible size is also wrong when the trigger is too large.

Reply:
I think the requirements indicate that the panel should always be flexible while also having a minimum width that can be modified for certain circumstances like the year picker.
- The panel should never have a smaller width than the trigger. 
- The panel should only be wider then the trigger if the trigger is less wide than the panel's minimum width.

As a result of these requirements I think the only api we need is `panelMinimumWidth`. It should default to 160, but could be changed for a year picker or some other extenuating circumstance.

Reply:
> As a result of these requirements I think the only api we need is panelMinimumWidth. It should default to 160, but could be changed for a year picker or some other extenuating circumstance.

Hmm, so if we always want the API to be flexible and only allow changing the minimum size or `width` I think it would have the same effect as the minimum, it makes sense so the panel always follows the size of the trigger with the `width` clamp with the default value of `160px`. Okay, this seems simpler.

Reply:
I sent new changes, it looks better now I also added more examples in the storybook to exemplify the use cases. Let me know what you guys think.

Reply:
This looks good! Thanks @matuzalemsteles ... After fixing the two comments left by @pat270 and myself it should be ready to be merged. 😄